docs/docs/typescript/generated-types.md

```diff
@@ -0,0 +1,157 @@
+---
+description: A look at automatic type generation in Redwood
+---
+
+# Generated Types
+
+To add to the TypeScript (and JavaScript!) experience, Redwood generates types for you.
+These generated types not only include your GraphQL operations, but also your named routes, Cells, scenarios, and tests.
+
+When you run `yarn rw dev`, the CLI watches files for changes and triggers type generation automatically, but you can trigger it manually too:
+
+```
+yarn rw g types
+```
+
+:::tip Getting errors trying to generate types?
+
+If you're getting errors trying to generate types, it's worth checking the GraphQL operations in your Cells and SDLs.
+Make sure that they're syntactically valid, and that every query and mutation on the web side is defined in an `*.sdl.js` file on the api side.
+
+:::
+
+If you're curious, you can find the generated types in the `.redwood/types`, `web/types/graphql.d.ts`, and `api/types/graphql.d.ts` directories. Broadly speaking, Redwood generates the following types:
+
+1. ["mirror" types](https://www.typescriptlang.org/docs/handbook/module-resolution.html#virtual-directories-with-rootdirs) for your components, Cells, and layouts on the web side
+2. types based on your queries and mutations on the web side (in `web/types/graphql.d.ts`)
+3. types for resolvers based on your SDLs on the api side (in `api/types/graphql.d.ts`)
+4. types for testing, `currentUser`, etc.
+
+## CurrentUser
+
+If you've setup auth, the type for the current user on both the web and the api side gets automatically "inferred" from the `getCurrentUser` function in `api/src/lib/auth.ts`.
+
+For example, if you specify the return type on `getCurrentUser` as...
+
+```ts title="api/src/lib/auth.ts"
+interface MyCurrentUser {
+  id: string,
+  roles: string[],
+  email: string,
+  projectId: number
+}
+
+const getCurrentUser = ({decoded}): MyCurrentUser => {
+  //..
+}
+```
+
+The types for both `useAuth().currentUser` on the web side and `context.currentUser` on the api side will be the same—the `MyCurrentUser` interface.
```

Comment:
I updated the copy—is this still correct?

Reply:
Ummm.. I think its important that we point out that the type _comes_ from the `getCurrentUser` function

Reply:
Actually, I covered it above. Change looks good